thirdparty/quicknode.go

```diff
@@ -60,17 +60,31 @@ func (v *QuicknodeVendor) GetVendorSpecificErrorIfAny(resp *http.Response, jrr i
 				common.NewErrJsonRpcExceptionInternal(code, common.JsonRpcErrorCapacityExceeded, msg, nil, details),
 			)
 		} else if strings.Contains(msg, "failed to parse") {
+			// We do not retry on parse errors, as retrying another upstream would not help.
 			return common.NewErrEndpointClientSideException(
 				common.NewErrJsonRpcExceptionInternal(code, common.JsonRpcErrorParseException, msg, nil, details),
+				false, // not retryable
 			)
-		} else if code == -32010 {
+		} else if code == -32010 { // Transaction cost exceeds current gas limit
+			// We do not retry low gas limit errors, as retrying another upstream would not help.
```

Comment:
If i'm not wrong the gas limit is a node-specific config, if that's true then trying on another upstream might be actually a correct behavior

Reply:
This indeed seems to be from [Parity](https://github.com/openethereum/parity-ethereum/blob/55c90d4016505317034e3e98f699af07f5404b63/rpc/src/v1/helpers/errors.rs#L452)

Reply:
I mean the "max gas" is probably a node-specific config (geth, reth, etc), based on this code it seems to be: https://github.com/openethereum/parity-ethereum/blob/v2.7.2-stable/miner/src/pool/verifier.rs#L174

Reply:
seems like it. if I understood correctly, each client can configure their own `gas limit` and also always make sure it is always lower than the `block gas limit overall`.

So one client (e.g reth) might have configured a gas limit higher than parity (but still below block gas limit) and therefore retrying another upstream would make sense under this logic.